subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/GroovySourceIncrementalCompilationIntegrationTest.groovy

```diff
@@ -156,17 +156,24 @@ class A2{}
     def 'merge old class source mappings if no recompilation required'() {
```

Comment:
This was testing an implementation detail. I've fixed it to test the resulting behavior instead.